designs/disruption-controls-by-action.md

```diff
@@ -0,0 +1,218 @@
+# Disruption Controls By Action 
+
+## Motivation
+Users want the ability to have different disruption controls for the budgets. The AKS Provider Drives NodeImageUpgrade, and K8sVersionUpgrade through Drift. Users may want to specify different disruption settings for upgrade, since upgrade is a fundementally different type of disruption in comparison to consolidation for example. Since upgrade changes the behavior of the cluster, its disruption schedule may be different from consolidation.
```

Comment:
Oh interesting...how do you do K8s version upgrade through drift? I'd assume that that's a control-plane concept so how does Karpenter know that it should auto-upgrade the K8s version?

Reply:
For k8s version upgrade it just checks through api-server, for node image it just checks for 1. CRD Changes(we allow specifying node image) or 2. There is a new version available if they have autoupgrade style turned on. 

We should do a full walkthrough with the community of both codebases, the AKS Codebase and the AWS Codbase and core.

Reply:
https://github.com/Azure/karpenter/blob/main/pkg/cloudprovider/drift.go#L72 we have these drift functions, and we mark IsDrifted if any of them return to do so. 
https://github.com/Azure/karpenter/blob/main/pkg/cloudprovider/cloudprovider.go#L190